src/io/flutter/bazel/WorkspaceCache.java

```diff
@@ -129,7 +132,12 @@ public void unsubscribe(Runnable callback) {
    */
   private void refresh() {
     final Workspace workspace = Workspace.loadUncached(project);
-    if (workspace == cache) return;
+    if (workspace == cache && !disconnected) return;
+    if (cache != null && workspace == null) {
+      disconnected = true;
```

Comment:
should we instead return early here after setting 
`disconnected = true;`
? 
setting cache=null in this case seems a bit odd.

Reply:
I thought it would be better for cache to be null when going on to notify listeners, because the only listener we want to behave differently is the workspace itself so that it keeps watching the previous file. For devices I think we still want the menu to disappear and for developers to not be able to run a project if they're disconnected.